src/sources/kubernetes_logs/mod.rs

```diff
@@ -734,14 +759,48 @@ impl Source {
         )
         .backoff(watcher::DefaultBackoff::default());
 
+        // Create shared broadcast channel for pod events
+        let (pod_event_tx, _) = tokio::sync::broadcast::channel(1000);
+        let reflector_rx = pod_event_tx.subscribe();
+
+        // Spawn task to forward pod events to broadcast channel
+        let pod_forwarder_tx = pod_event_tx.clone();
+        let pod_forwarder = tokio::spawn(async move {
+            pin!(pod_watcher);
+            while let Some(event_result) = pod_watcher.next().await {
+                match event_result {
+                    Ok(event) => {
+                        // Only broadcast successful events
+                        if pod_forwarder_tx.send(event).is_err() {
+                            // All receivers have been dropped
+                            break;
+                        }
+                    }
+                    Err(e) => {
+                        warn!("Pod watcher error: {}", e);
+                        // Continue on errors to maintain resilience
+                    }
+                }
+            }
+        });
+        reflectors.push(pod_forwarder);
+
+        // Convert broadcast receiver to stream for reflector
+        let reflector_stream = BroadcastStream::new(reflector_rx).filter_map(|result| {
+            ready(match result {
+                Ok(event) => Some(Ok(event)),
+                Err(_) => None,
```

Comment:
Silently dropping the error here is a bad idea. I have recently come across an issue (#24220) where a broadcast channel is erroring due to RecvError::Lagged. This will silently drop logs.

I am weary of using a broadcast channel here for the same reason as listed in that issue. IMO a remediation needs to be explored for 24220 and the same strategy needs to be applied here